OpenRVS/classes/OpenServerList.uc

```diff
@@ -89,128 +59,36 @@ event ReceivedText(string Text)
 //a server name",IP="44.434.4.434:7777",Locked=false,GameMode="Adver
 //the function in OpenMultiPlayerWidget will parse this further to get the needed info.
 //1.3 - NO LONGER parsed in OpenMultiPlayerWidget, parsed below
-event Closed()
+function ParseServers(OpenHTTPClient.HttpResponse resp)
 {
-	local int i;//index for counting received strings
-	local int j,k;
-	local array<string> list;//array to send to widget
-	local string s;//currently working with
-	local string l;//the LAST complete server info we found
-	class'OpenLogger'.static.DebugLog("CLOSED");
-	super.Closed();
-	class'OpenLogger'.static.DebugLog("lines received: " $ buffer.length);
-	//THE FOLLOWING:
-	//takes any data received from the web address
-	//and splits it into chunks smaller than or equal to 85 characters.
-	//these chunks are easily manageable and can be manipulated and parsed easily.
-	//when strings are too large it can cause issues or crash the game
-	//so to work with the received data we break it apart.
-	//85 chosen because you can add up to 3 strings of that size together and will still be less than 256 chars.
-	//don't know if this actually matters lol
-	i = 0;
-	while ( i < buffer.length )
+	local array<string> lines;
+	local string line;
+	local int i;
+
+	//Check for errors.
+	if (resp.Code != 200)
 	{
-		class'OpenLogger'.static.DebugLog("line " $ i+1 $ " length: " $ len(buffer[i]));
-		while ( len(buffer[i]) > 85 )
-		{
-			class'OpenLogger'.static.DebugLog("moving 85 characters to next array element");
-			buffer.insert(i+1,1);
-			buffer[i+1] = right(buffer[i],85);
-			ReplaceText(buffer[i],buffer[i+1],"");//get rid of the text we moved to next array element
-		}
-		class'OpenLogger'.static.DebugLog("line " $ i $ ": " $ buffer[i]);
-		i++;
+		class'OpenLogger'.static.Error("failed to retrieve servers over http", self);
+		NoServerList();
 	}
-	//get rid of extraneous received data before the serverlist
-	//NOTE: shouldn't be an issue but assumes that there's at least two entries in buffer array
-	s = buffer[0] $ buffer[1];
-	while ( InStr(s,"ServerName=") == -1 )
+
+	// Convert each line to CSV K=V format (i.e. "a=b,c=d,e=f")
+	class'OpenString'.static.Split(resp.Body, Chr(10), lines);
+	for (i=1; i<lines.Length; i++)//skips first line

```

Comment:
Just so I'm clear -- we're skipping the first line because it's assumed it's the [OpenRVS.OpenMultiPlayerWidget] line right?

Reply:
That's right - this Parse function is specific to the current format. We can easily add another Parse function now for CSV format!